packages/gatsby/src/utils/get-latest-apis.ts

```diff
@@ -1,20 +1,24 @@
-const path = require(`path`)
-const fs = require(`fs-extra`)
-const axios = require(`axios`)
+import path from "path"
+import fs from "fs-extra"
+import axios from "axios"
 
 const API_FILE = `https://unpkg.com/gatsby/apis.json`
 const ROOT = path.join(__dirname, `..`, `..`)
 const OUTPUT_FILE = path.join(ROOT, `latest-apis.json`)
 
-module.exports = async function getLatestAPI() {
+export interface IAPIResponse {
+  [key: string]: any
+}
+
+export const getLatestAPIs = async (): Promise<IAPIResponse> => {
   try {
     const { data } = await axios.get(API_FILE, { timeout: 5000 })
 
     await fs.writeFile(OUTPUT_FILE, JSON.stringify(data, null, 2), `utf8`)
 
     return data
   } catch (e) {
-    if (await fs.exists(OUTPUT_FILE)) {
+    if (await fs.existsSync(OUTPUT_FILE)) {
```

Comment:
I think we should use a sync method.

Reply:
https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/pathExists-sync.md

Reply:
How come? I don't see the benefit, but happy to be convinced!

Reply:
We await for the async one, so we could use the sync one instead.
But need to delete the `await` key then

Reply:
Still want to understand why we want the sync one. This does put a different expense on the system. When you await things, you allow the node process to handle other things on the stack that are paused. I'm not seeing the benefit of doing sync here.

Reply:
Sorry, when I search for `fs-extra` I found out that `fs.exist(OUTPUT_FILE)` returns a promise. So I shouldn't change to `existSync`. :bow:
> `fs-extra` is a drop in replacement for native `fs`. All methods in fs are attached to `fs-extra`. **All fs methods return promises if the callback isn't passed.**
[ref](https://www.npmjs.com/package/fs-extra#usage)

However, this is a bit troublesome because TS infers `fs.exists` as native `fs.exists`. The error `An argument for 'callback' was not provided.` still remains.
```ts
function exists(path: PathLike, callback: (exists: boolean) => void): void;
```
And be careful that `exist()` of native `fs` returns void, not a promise and it needs a callback function on second argument. And this is [deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback).

Reply:
looks like we just want this instead: https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md

if this doesn't work, I think the best thing would be to submit a PR to update the types in DefinitelyTyped